components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/device/dao/DeviceFlowDAOImpl.java

```diff
@@ -0,0 +1,452 @@
+import org.wso2.carbon.identity.oauth2.device.constants.Constants;
+import org.wso2.carbon.identity.oauth2.device.model.DeviceFlowDO;
+import org.wso2.carbon.identity.oauth2.util.OAuth2Util;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Timestamp;
+import java.util.ArrayList;
+import java.util.Calendar;
+import java.util.Date;
+import java.util.List;
+import java.util.TimeZone;
+import java.util.UUID;
+
+/**
+ * This class contains override methods of DeviceFlowDAO.
+ */
+public class DeviceFlowDAOImpl implements DeviceFlowDAO {
+
+    private static final Log log = LogFactory.getLog(DeviceFlowDAOImpl.class);
+
+    private String clientId;
+    private String status;
+    private String scope;
+
+    @Override
+    public void insertDeviceFlowParameters(String deviceCode, String userCode, String consumerKey, Long expiresIn,
+                                 int interval) throws IdentityOAuth2Exception {
+
+        if (log.isDebugEnabled()) {
+            log.debug("Persisting device_code: " + deviceCode + "for client: " + consumerKey);
+        }
+        try (Connection connection = IdentityDatabaseUtil.getDBConnection(true)) {
+            try (PreparedStatement prepStmt =
+                         connection.prepareStatement(SQLQueries.DeviceFlowDAOSQLQueries.STORE_DEVICE_CODE)) {
+                Date date = new Date();
+                Timestamp timeCreated = new Timestamp(date.getTime());
+                long timeExpired = timeCreated.getTime() + expiresIn;
+                Timestamp expiredTime = new Timestamp(timeExpired);
+                prepStmt.setString(1, UUID.randomUUID().toString());
+                prepStmt.setString(2, deviceCode);
+                prepStmt.setString(3, userCode);
+                prepStmt.setTimestamp(4, timeCreated, Calendar.getInstance(TimeZone
+                        .getTimeZone(Constants.UTC)));
+                prepStmt.setTimestamp(5, timeCreated, Calendar.getInstance(TimeZone
+                        .getTimeZone(Constants.UTC)));
+                prepStmt.setTimestamp(6, expiredTime, Calendar.getInstance(TimeZone
+                        .getTimeZone(Constants.UTC)));
+                prepStmt.setLong(7, interval);
+                prepStmt.setString(8, Constants.PENDING);
+                prepStmt.setString(9, consumerKey);
+                prepStmt.execute();
+                IdentityDatabaseUtil.commitTransaction(connection);
+            } catch (SQLException e) {
+                IdentityDatabaseUtil.rollbackTransaction(connection);
+                throw new IdentityOAuth2Exception("Error when storing the device flow parameters for consumer_key: " +
+                        consumerKey, e);
+            }
+        } catch (SQLException e) {
+            throw new IdentityOAuth2Exception("Error when storing the device flow parameters for consumer_key: " +
+                    consumerKey, e);
```

Comment:
Both the error messages give the same message. Shall we give more context information to each message

Reply:
The purpose of the upper catch box is to get rollback connection. Otherwise, errors are the same.